client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-recipe.ts

```diff
@@ -51,13 +52,10 @@ const useRecipe = ( siteId = 0, patterns: Pattern[], categories: Category[] ) =>
 						category,
 					};
 				} ),
-		/* We initialize the sections (at most) once when the patterns and categories are finally loaded
-		   (hence the ".length"s in the dependency array).
-
-		   We are ignoring the changes from the search params (section_pattern_ids),
+		/* We are ignoring the changes from the search params (section_pattern_ids),
 	       since after this point, the changes will be handled by setSections().
 	    */
-		[ patterns.length, categories.length ] // eslint-disable-line react-hooks/exhaustive-deps
+		[ patternsById, categoriesByName ] // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
[nit] We might not need to disable eslint here

Reply:
We need because we're ignoring `section_pattern_ids`

Reply:
Oh, got it!